src/Mod/Part/App/Geometry.cpp

```diff
@@ -6244,6 +6244,419 @@ GeomArcOfCircle *createFilletGeometry(const GeomLineSegment *lineSeg1, const Geo
+                interpoints = std::make_pair(tmpp1, tmpp2);
+                pos1 = 1;
+                pos2 = 1;
+            }
+        }
+        else if ((curve1->getEndPoint() - curve2->getStartPoint()).Length() < Precision::Confusion()) {
+            Base::Vector3d tmpp1 = curve1->getEndPoint();
+            Base::Vector3d tmpp2 = curve2->getStartPoint();
+            double tmpdist = distancetorefpoints(tmpp1, tmpp2, refPnt1, refPnt2);
+            if (tmpdist < dist) {
+                dist = tmpdist;
+                interpoints = std::make_pair(tmpp1, tmpp2);
+                pos1 = 2;
+                pos2 = 1;
+            }
+        }
+        else if ((curve1->getStartPoint() - curve2->getEndPoint()).Length() < Precision::Confusion()) {
+            Base::Vector3d tmpp1 = curve1->getStartPoint();
+            Base::Vector3d tmpp2 = curve2->getEndPoint();
+            double tmpdist = distancetorefpoints(tmpp1, tmpp2, refPnt1, refPnt2);
+            if (tmpdist < dist) {
+                dist = tmpdist;
+                interpoints = std::make_pair(tmpp1, tmpp2);
+                pos1 = 1;
+                pos2 = 2;
+            }
+        }
+        else if ((curve1->getEndPoint() - curve2->getEndPoint()).Length() < Precision::Confusion()) {
+            Base::Vector3d tmpp1 = curve1->getEndPoint();
+            Base::Vector3d tmpp2 = curve2->getEndPoint();
+            double tmpdist = distancetorefpoints(tmpp1, tmpp2, refPnt1, refPnt2);
+            if (tmpdist < dist) {
+                dist = tmpdist;
+                interpoints = std::make_pair(tmpp1, tmpp2);
+                pos1 = 2;
+                pos2 = 2;
+            }
+        }
+
+        if (dist == INFINITY) {
+            // no coincident was found, try basis curve intersection if GeomTrimmedCurve
+            if (geo1->isDerivedFrom<GeomTrimmedCurve>() && geo2->isDerivedFrom<GeomTrimmedCurve>()) {
+                auto* tcurve1 =static_cast<const GeomTrimmedCurve*>(geo1);
+                auto* tcurve2 = static_cast<const GeomTrimmedCurve*>(geo2);
+
+                try {
+                    if (!tcurve1->intersectBasisCurves(tcurve2, points)) {
+                        return nullptr;
+                    }
+                }
+                catch (Base::CADKernelError& e) {
+                    e.ReportException();
+                    THROWMT(Base::CADKernelError,
+                            QT_TRANSLATE_NOOP("Exceptions",
+                                              "Unable to guess intersection of curves. Try adding "
+                                              "a coincident constraint between the vertices of the "
+                                              "curves you are intending to fillet."))
+                }
+
+                int res = selectintersection(points, interpoints, refPnt1, refPnt2);
+
+                if (res != 0) {
+                    return nullptr;
+                }
+            }
+            else {

```

Comment:
Since you are refactoring, please try to reduce the cognitive complexity where possible. Here, you can switch the if and else parts and avoid one level of nesting.